calyx/src/frontend/ast.rs

```diff
@@ -18,6 +18,8 @@ pub struct NamespaceDef {
     pub components: Vec<ComponentDef>,
     /// Extern statements and any primitive declarations in them.
     pub externs: Vec<(String, Vec<ir::Primitive>)>,
+    /// Inlined Primitives (i.e., primitives with verilog definitions)
+    pub prim_inlines: Vec<ir::Primitive>,
```

Comment:
Is there a way to merge the `externs` field with `prim_inlines`? Perhaps by using `Vec<Option<String>, ...>` to indicate the primitives that don't have a parent file need to be inlined?

Reply:
Do you think I should do a similar thing for `ctx.lib` (i.e., having one `Vec<Option,_>` instead of two separate vecs? In some ways, having two separate vecs in the `Context` is convenient, but in other scenarios having it as one `vec` is better.